Initial runthrough for clarity (#109)
nfidd/nfidd: sessions/biases-in-delay-distributions.qmd

library("cmdstanr")
```

> ::: callout-tip

Reviewer: I had no idea this works. I was going by the instructions in https://quarto.org/docs/authoring/callouts.html

head(df_dates)
```

> ::: callout-note

Reviewer: ```suggestion
::: {.callout-note}
```

As before we are still not working with dates but numbers.
This makes handling the data easier - we don't have to make any conversions before using the data in stan.
:::

Each of the numbers now represent the number of days that have passed since the start of the outbreak.
That is, each of the numbers correspond to a day.

> In that sense, the data is more like typical data we get from infectious disease outbreaks.

Reviewer: ```suggestion
In that sense, the data is more like typical data we get from infectious disease outbreaks, where we would usually have a line list with key events such as symptom onset or death reported by a *date*.
```

> However, we have now *censored* the data.

Reviewer: ```suggestion
That is, the data is *censored*.
```

## Estimating delay distributions accounting for censoring

Let's estimate the time from symptom onset to hospitalisation with the censored data.

A naïve approach to estimating the delay would be to ignore the fact that the data are censored.

> To estimate the delay from onset to hospitalisation, we could just use the integer difference in days.

Reviewer: ```suggestion
To estimate the delay from onset to hospitalisation, we could just use the difference between the censored times, which is an integer (the number of days).
```

:::

To account for censoring, we need to modify the model to include the fact that we don't know when exactly on any given day the event happened.
For example, if we know that symptom onset of an individual occurred on 20 June, 2024, and they were admitted to hospital on 22 June, 2024, this could mean an onset-to-hospitalisation delay from 1 day (onset at 23:59 on the 20th, admitted at 0:01 on the 22nd) to 3 days (onset at 0:01 on the 20th, admitted at 23:59 on the 22nd).

We can use this in our delay estimation by making the exact time of the events based on the dates given part of the estimation procedure:

```{r censoring_adjusted_delay_model}
cmod <- cmdstan_model(here("stan", "censored_delay_model.stan"))
cmod$print(line_numbers = TRUE)
```

::: callout-tip
## Take 5 minutes

Familiarise yourself with the model above.
Do you understand all the lines?
Which line(s) define the parameter prior distribution(s), which one(s) the likelihood, and which one(s) reflect that we have now provided the delay as the difference in integer days?
:::

::: {.callout-note collapse="true"}
## Solution

> Lines 21-24 define the parametric prior distributions (for parameters meanlog and sdlog, and the estimates of exact times of events).

Reviewer: ```suggestion
Lines 22-25 define the parametric prior distributions (for parameters meanlog and sdlog, and the estimates of exact times of events).
```

cres
```

::: callout-tip
## Take 10 minutes

Try re-simulating the delays using different parameters of the delay distribution.
Can you establish under which conditions the bias in estimation gets worse?
:::

## Real-time estimation: truncation

The data set we have looked at so far in this session is a "final" data set representing an outbreak that has come and gone.
However, information on delay distribution is often important during ongoing outbreaks as they can inform nowcasts and forecasts and help with broader interpretation of data.

> Estimating delays in real time comes with particular challenges, as the timing of the cut-off might introduce a bias.

Reviewer: ```suggestion
Estimating delays in real time comes with particular challenges, as the timing of when a snapshot of data was used for analysis might introduce a bias.
```

If, for example, infections are exponentially increasing then there will be disproportionately more people with recent symptom onset.

> This would artificially decrease the estimate of the mean delay compared to its true value for all infections.

Reviewer: ```suggestion
Without adjustment, this would artificially decrease the estimate of the mean delay compared to its true value for all infections.
```
